pkg/providers/v2/loadbalancer.go

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v2 is an out-of-tree only implementation of the AWS cloud provider.
+// It is not compatible with v1 and should only be used on new clusters.
+package v2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	v1 "k8s.io/api/core/v1"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/klog/v2"
+)
+
+// newLoadBalancer returns an implementation of cloudprovider.LoadBalancer
+func newLoadBalancer(region string, creds *credentials.Credentials) (cloudprovider.LoadBalancer, error) {
```

Comment:
@andrewsykim I'm just following the pattern for implementing the LoadBalancer interface
For the functions to implement, I leave EnsureLoadBalancer() and DeleteLoadBalancer() empty and still try to gain a better understanding of the implementation
For other functions, I still have a lot of TODOs and probably need more contexts, so I might change my code a lot very soon, this just shows the basic structure of my implementation